### PR TITLE
[CELEBORN-1375] Introduce worker module for dashboard frontend

### PR DESCRIPTION
**Pull request by tiny-dust**
### What changes were proposed in this pull request?

Introduce worker module for dashboard frontend:
image

### Why are the changes needed?

The worker page provides Celeborn with worker lists and overview information.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Local test.

### CODE REVIEW

File: web/src/views/worker/components/WorkerTable.vue [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    key: 'pushPort'
+  },
+  {
+    title: 'ReplicatePort',
+    key: 'replicatePort'
+  },
+  {
+    title: 'Version',
+    key: 'version'
+  }
```
Comment (SteNicholas) [inline web/src/views/worker/components/WorkerTable.vue:83]: Where is the `More` field?

File: web/.vscode/extensions.json [hunk 2]
```diff
@@ -1,7 +1,6 @@
 {
   "recommendations": [
     "Vue.volar",
-    "Vue.vscode-typescript-vue-plugin",
```
Comment (SteNicholas) [inline web/.vscode/extensions.json:4]: Why remove this extension?

File: web/src/views/worker/components/WorkerForm.vue [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+const reset = () => {
+  formData.master = ''
+  emits('reset', unref(formData))
+}
+</script>
+
+<template>
+  <n-form :model="formData" :show-feedback="false" label-placement="left" @keydown.enter="search">
+    <n-grid :x-gap="24" :y-gap="24" :cols="4">
+      <n-form-item-gi label="Master" path="master">
```
Comment (SteNicholas) [inline web/src/views/worker/components/WorkerForm.vue:39]: Could we support search with other fields like `HostName` etc?

File: web/src/views/worker/components/WorkerForm.vue [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        <n-input v-model:value="formData.master" placeholder="" clearable />
+      </n-form-item-gi>
+      <n-grid-item>
+        <n-flex>
+          <n-button type="primary" ghost @click="search">Search</n-button>
```
Comment (SteNicholas) [inline web/src/views/worker/components/WorkerForm.vue:44]: Could the name of the click function align with the application module?

File: web/src/views/worker/components/WorkerForm.vue [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+          <n-button ghost @click="reset">Reset</n-button>
```
Comment (SteNicholas) [inline web/src/views/worker/components/WorkerForm.vue:45]: Ditto.

**Review by SteNicholas — CHANGES_REQUESTED** [review]

File: web/.vscode/extensions.json (hunk 2 above)
Comment (tiny-dust) [reply web/.vscode/extensions.json:4]: Plugin has been deprecated.

File: web/src/views/worker/components/WorkerForm.vue (hunk 3 above)
Comment (tiny-dust) [reply web/src/views/worker/components/WorkerForm.vue:39]: which fields specifically need to be searched?

File: web/src/composables/pagination.ts [hunk 6]
```diff
@@ -16,91 +16,85 @@
...
-  async function search(newSearch?: SearchForm) {
-    pageNum.value = 1
-    await updatePaginationState(newSearch)
+  function doSearch(formData: Record<string, string>) {
+    currentPage.value = 1
+    searchParams.value = formData
+    toLoadData()
+  }
+
+  async function toLoadData() {
```
Comment (SteNicholas) [inline web/src/composables/pagination.ts:73]:
```suggestion
  async function loadData() {
```

File: web/src/views/application/components/ApplicationSearchForm.vue [hunk 7]
```diff
@@ -17,24 +17,19 @@
...
-
-const $emits = defineEmits(['do-search', 'do-reset'])
-
-const doSearch = () => {
-  $emits('do-search', unref(searchForm))
-}
+const searchForm = reactive({
+  appId: '',
+  subUser: '',
+  tenant: ''
```
Comment (SteNicholas) [inline web/src/views/application/components/ApplicationSearchForm.vue:29]: Add state?

**Pull request closed without merging**